modules/library/metadata/src/main/java/org/geotools/util/factory/GeoTools.java

```diff
@@ -133,8 +133,7 @@ private static Properties loadProperites(String resource) {
      * The bindings between {@linkplain System#getProperties system properties} and a hint key. This
      * field must be declared before any call to the {@link #bind} method.
      */
-    private static final Map<String, RenderingHints.Key> BINDINGS =
-            new HashMap<String, RenderingHints.Key>();
+    private static final Map<String, RenderingHints.Key> BINDINGS = new HashMap<>();
```

Comment:
So... the synchronizations are not needed because BINDINGS is modified only during the class initialization itself (at least, with the current code). Right?

Reply:
@groldan can you confirm/elaborate?

Reply:
Hi @aaime yes, I think the commit message states it pretty clearly, let me know if it's confusing anyhow so I can rephrase:
```
GeoTools.scanForSystemHints() synchronizes on its static
BINDINGS variable, which is only modified during the class'
static initialization. The method is side-effect free and
only mutates the argument object, hence there's no need
to synchronize on this field.
```

Reply:
well,  `GeoTools.scanForSystemHints(final Map<RenderingHints.Key, Object> hints)` is not "side effect free" because it mutates the argument object, I `BINDINGS` is never modified once the class is initialized.